SU2_CFD/src/solvers/CSolver.cpp

```diff
@@ -4265,74 +4268,148 @@ void CSolver::LoadInletProfile(CGeometry **geometry,
-              cout << "]" << endl;
-              cout << "Distance to closest point: " << min_dist << endl;
-              cout << "Current tolerance:         " << tolerance << endl;
-              cout << endl;
-              cout << "You can widen the tolerance for point matching by changing the value" << endl;
-              cout << "of the option INLET_MATCHING_TOLERANCE in your *.cfg file." << endl;
-              local_failure++;
-              break;
+                    /*--- Check is this is the closest point and store data if so. ---*/
+
+                    if (dist < min_dist) {
+                    min_dist = dist;
+                    for (iVar = 0; iVar < nColumns; iVar++)
+                        Inlet_Values[iVar] = Inlet_Data[index+iVar];
+                    }
+
+                }
+
+                /*--- If the diff is less than the tolerance, match the two.
+                We could modify this to simply use the nearest neighbor, or
+                eventually add something more elaborate here for interpolation. ---*/
+
+                if (min_dist < tolerance) {
 
+                    solver[MESH_0][KIND_SOLVER]->SetInletAtVertex(Inlet_Values.data(), iMarker, iVertex);
+
+                } else {
+
+                    unsigned long GlobalIndex = geometry[MESH_0]->node[iPoint]->GetGlobalIndex();
+                    cout << "WARNING: Did not find a match between the points in the inlet file" << endl;
+                    cout << "and point " << GlobalIndex;
+                    cout << std::scientific;
+                    cout << " at location: [" << Coord[0] << ", " << Coord[1];
+                    if (nDim ==3) error_msg << ", " << Coord[2];
+                    cout << "]" << endl;
+                    cout << "Distance to closest point: " << min_dist << endl;
+                    cout << "Current tolerance:         " << tolerance << endl;
+                    cout << endl;
+                    cout << "You can widen the tolerance for point matching by changing the value" << endl;
+                    cout << "of the option INLET_MATCHING_TOLERANCE in your *.cfg file." << endl;
+                    local_failure++;
+                    break;
+                }
+
+            }
+
+            else if(Interpolate == true){
+              /* --- Calculating the radius and angle of the vertex ---*/
+              Interp_Radius = sqrt(pow(Coord[0],2)+ pow(Coord[1],2));
+              Theta = atan2(Coord[1],Coord[0]);
+
+              /* --- Evaluating and saving the final spline data ---*/
+              for  (unsigned short iVar=0; iVar < nColumns; iVar++){
+              //Evaluate spline will get the respective value of the Data set (column) specified
+              //for that interpolator[iVar], cycling through all columns to get all the 
+              //data for that vertex
+                Inlet_Interpolated[iVar]=interpolator[iVar]->EvaluateSpline(Interp_Radius);
```

Comment:
I call it like this now without passing the object in the argument but I use `this->` pointer to keep track of which interpolation class instance I am dealing with.